atlas-json-parent/atlas-json-core/src/main/java/io/atlasmap/json/inspect/SchemaInspector.java

```diff
@@ -0,0 +1,124 @@
+package io.atlasmap.json.inspect;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.atlasmap.json.v2.AtlasJsonModelFactory;
+import io.atlasmap.json.v2.JsonComplexType;
+import io.atlasmap.json.v2.JsonDocument;
+import io.atlasmap.json.v2.JsonFields;
+import io.atlasmap.v2.CollectionType;
+import io.atlasmap.v2.FieldStatus;
+import io.atlasmap.v2.FieldType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Iterator;
+import java.util.Map.Entry;
+
+/**
+ */
+public class SchemaInspector {
+
+    private JsonDocument jsonDocument = AtlasJsonModelFactory.createJsonDocument();
+    private static final Logger logger = LoggerFactory.getLogger(SchemaInspector.class);
+
+    public SchemaInspector() {
+    }
+
+    public JsonDocument getJsonDocument() {
+        return jsonDocument;
+    }
+
+    public void inspect(String schema) throws JsonInspectionException {
```

Comment:
What about making the inspect atomic with respect to generating the JsonDocument (would make it thread-safe, too). I.e. having a `JsonDocument inspect(String schema)` where the document itself is created within the method ?  The method then could even be `static`.